src/com/spikes2212/genericsubsystems/limitationFunctions/TwoLimits.java

```diff
@@ -0,0 +1,53 @@
+package com.spikes2212.genericsubsystems.limitationFunctions;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import com.spikes2212.genericsubsystems.BasicSubsystem;
+
+/**
+ * This is a Function from double to boolean. This is for a
+ * {@link BasicSubsystem} that moves between two limits.
+ * 
+ * @author Omri "Riki" Cohen
+ *
+ */
+public class TwoLimits implements Function<Double, Boolean> {
+
+	private Supplier<Boolean> maxLimit, minLimit;
+
+	/**
+	 * Constructs a two limits function using 2 boolean suppliers.
+	 * 
+	 * @param maxLimit
+	 *            the upper limit, positive speed makes the
+	 *            {@link BasicSubsystem} move towards this limit.
+	 * @param minLimit
+	 *            the lower limit, negative speed makes the
+	 *            {@link BasicSubsystem} move towards this limit.
+	 */
+	public TwoLimits(Supplier<Boolean> maxLimit, Supplier<Boolean> minLimit) {
+		this.minLimit = maxLimit;
+		this.maxLimit = minLimit;
+	}
+
+	/**
+	 * This get a speed and checks if the max limit is pressed and if a positive
+	 * speed is given or if the min limit is pressed and if a negative speed is

```

Comment:
your doc was very specific - which is good. but I want you to add more general stuff, like "this method checks if the subsystem can move...", in addition to the existing docs